### PR TITLE
fix: [M3-8342] - Toast Notifications for Images 

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- Removes some unnecessary styles for our Toast implementation 🗑️ 
- Fixes flaw with `useToastNotifications` that caused toasts to show up with no message 🔧 
- Improves how content wraps inside of toast notifications 🎨 
- Adds a `maxWidth` to Toast Notifications on larger screens so that they don't take up the full width of the screen
  - See the `WithLongMessage` Storybook story 📖 

## Preview 📷

| Before  | After   |
| ------- | ------- |
| <video src="https://github.com/linode/manager/assets/115251059/4970469b-4215-4951-9843-aa6404c956ac" /> | <video src="https://github.com/linode/manager/assets/115251059/79742e20-8b9c-44ef-a51a-e2e9f8bae9cb" /> |
| <video src="https://github.com/linode/manager/assets/115251059/0aeef353-615a-4379-acc8-e9d66a5f71e2" /> | <video src="https://github.com/linode/manager/assets/115251059/0ee71317-3107-4120-805d-a8b6768617f6" /> |

## How to test 🧪

- Use `yarn storybook` and verify that all Toast variants look correct in both light 🌞  and dark mode 🌚 
- Test canceling an uploaded image that is the "processing" state ⛔️
- Test the error toast that shows when an Image fails to capture from a Linode 👀 
  - I'm currently able to reproduce this in dev when trying to capture an image from a powered on Alpine Linux Linode, but your milage may vary 

## As an Author I have considered 🤔

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/hooks/useToastNotifications.tsx [hunk 1]
```diff
@@ -230,34 +231,39 @@ export const useToastNotifications = (): {
...
       const failureMessage = getToastMessage(message, event);
-      const hasSupportLink =
-        failureMessage?.includes('contact Support') ?? false;
 
-      const formattedFailureMessage = createFormattedMessage(
-        failureMessage,
-        link,
-        hasSupportLink
-      );
+      if (failureMessage) {
```
Comment (bnussman-akamai) [inline packages/manager/src/hooks/useToastNotifications.tsx:252]: Just added `if` statements to ensure we only call `enqueueSnackbar` if the message is defined

File: packages/manager/src/components/Snackbar/Snackbar.tsx [hunk 2]
```diff
@@ -11,6 +10,9 @@ import type { SnackbarProviderProps } from 'notistack';
 
 const StyledMaterialDesignContent = styled(MaterialDesignContent)(
   ({ theme }: { theme: Theme }) => ({
+    '&.notistack-MuiContent': {
+      flexWrap: 'unset',
+    },
```
Comment (bnussman-akamai) [inline packages/manager/src/components/Snackbar/Snackbar.tsx:15]: The Snackbar container defaults to `flexWrap: 'wrap'`. We don't want the close button to wrap, so I added this to disable flex wrap.

**Review by jaalah-akamai — APPROVED** [review]
Thanks - looks good and the wrapping issue is now fixed ✅

**Review by hana-linode — APPROVED** [review]
Changes LGTM

Semi-related, is it expected that the toast background in light mode now matches the variant instead of being white? (Personally, I am not a fan of the colored background)

https://github.com/linode/manager/assets/115299789/002b70e7-0b11-4e61-b0ea-1f5c96d39dad

**Comment by jaalah-akamai** [comment]
> Semi-related, is it expected that the toast background in light mode now matches the variant instead of being white? (Personally, I am not a fan of the colored background)

Yea it's part of CDS 2.0

**Pull request merged**